server/controllers/request.js

```diff
@@ -0,0 +1,73 @@
+const Request = require("../models/Request");
+const Profile = require("../models/Profile");
+const asyncHandler = require("express-async-handler");
+
+// @route GET /request
+// @desc list of requests
+// @access Private
+exports.getRequests = asyncHandler(async (req, res, next) => {
+  const profile = await Profile.findOne({ userId: req.user.id });
+
+  if (profile.accountType === "pet_owner") {
+    const requests = await Request.find({ userId: req.user.id });
+
+    res.status(200).json({
+      success: {
+        requests: requests,
+      },
+    });
+  } else {
+    const requests = await Request.find({ sitterId: req.user.id });
+    res.status(200).json({
+      success: {
+        requests: requests,
+      },
+    });
+  }
+});
+
+// @route POST /request
+// @desc Create a new request
+// @access Private
+exports.makeRequest = asyncHandler(async (req, res, next) => {
+  const request = new Request(req.body.request);
+
+  if (!request.userId) {
```

Comment:
Nice to see validation! 

Optional: Alternatively, we can move this into a validation middleware to cleanup the code and increase maintainability